utils/wasm-gen/src/generator/entry_points.rs

```diff
@@ -99,9 +103,11 @@ impl<'a, 'b> EntryPointsGenerator<'a, 'b> {
     /// Returns disabled entry points generator and a proof that all entry points from config were generated.
     pub fn generate_entry_points(
         mut self,
+        mem_import_gen_proof: MemoryImportGenerationProof,
```

Comment:
make the proof to be a requirement for `EntryPointsGenerator` instantiation

Reply:
also adjust docs stating this

Reply:
I'm not sure if this can be done semantically correctly since `EntryPointsGenerator` is created from `GearWasmGenerator`. This means that we need to move `MemoryImportGenerationProof` to `GearWasmGenerator` as well.

Reply:
Done in the last two commits:
- now `generate_entry_points` is very similar to `generate_syscalls`
- new type `EntryPointsGeneratorInstantiator` has been introduced (`generate_syscalls` does that too)